Source/Core/Core/NetPlayServer.cpp

```diff
@@ -951,3 +1199,18 @@ std::vector<std::pair<std::string, std::string>> NetPlayServer::GetInterfaceList
     result.emplace_back(std::make_pair("!local!", "127.0.0.1"));
   return result;
 }
+
+bool NetPlay::IsHostAndSyncingSaveData()
+{
+  return netplay_server != nullptr && netplay_server->IsSyncingSaveData();
```

Comment:
Please don't add file-scoped variables for this sort of thing. Operate on the instance of the server instead or alter the interface of what you're trying to accomplish so that it accomodates that behavior (operating on the instance). This adds to the problem of global state in Core (even if it's static, it's effectively global behavior from a lifecycle perspective).

Reply:
Yes, I agree. That's one of the larger code smells that netplay as a whole has. Any non-const global state in core (or essentially any other library really), is generally not a good idea (and I don't think making that situation worse is a good idea either).

Reply:
Well, the instance of the client/server is located in Qt (`DolphinQt2/Settings.h`), and we can't depend on Qt in a core file, so this was the only way I could think to do it.

Reply:
Perhaps we just leave it this way for now, and address the issue of global state as a separate PR, since that's rather out of scope here?

Reply:
> Perhaps we just leave it this way for now, and address the issue of global state as a separate PR, since that's rather out of scope here?

This reads very much like, "a bad decision was made in the past, so another one will be made rather than correcting the first one, likely increasing the overall time to solve the issue as a whole since there's now two problems instead of one to deal with, but we can fix it later, for some unspecified definition of 'later', because it would delay [feature] that is trying to be added". I also apologize if that comes off wrong, but I've seen the suggestion of things being fixed later without them actually ending up being fixed quite a bit in the past.

Are you volunteering yourself as the one that will fix the global state issue with netplay, or is this just going to end up being left for someone else to fix?

Reply:
I understand your concerns. It's not that I want to just add another global state and not care, in fact I questioned why it was global in the first place (the instances don't really belong in Qt, now do they?), but in order to *not* add a global state here I would also have to fix all the global state issues.

I don't believe that's within the scope of this PR, but if you don't mind me going a bit out of scope here I can address it. Otherwise, I'll leave this as-is and address it in a separate PR.

Reply:
Wait, since these functions are static in the NetPlay namespace, isn't this effectively global regardless of the variable? Maybe I'm not fully understanding what you mean by global here.

Reply:
> Wait, since these functions are static in the NetPlay namespace

Which function(s)? This comment chain is taking place on a non-static file-scope function.

By global, I mean global variables (a static file-scoped variable is a form of global variable, since it's present for the entire duration of the program). 7226 adds a namespace, which gets things out of the unqualified global namespace (this is unrelated to whether or not a variable is considered a global variable. A global variable in a namespace is just a global variable with a qualifier on it).

Whether or not a function is file-scope static or not has no bearing on whether or not a behavior is global (static member functions of classes on the other hand, is a different story).

Reply:
Okay, I see. So then how exactly do we eliminate the global state? Move the instances out of Qt and into Core? Where exactly should they go and how should they be accessed? Because, as I understand it, lots of things in Dolphin are global (configuration for example).

Reply:
There's multiple ways. All of which require reworking code to fit/work with interfaces better, and depend on which state is being fixed. In the easiest case, it's just a matter of changing file-scope variables into class variables, and that's it.

In the case of NetPlay, without having looked at the code for a bit, one potential solution would be that the instances would live in the frontend, since that's where they're created. They don't belong in core, since not everyone that starts up the core also wants to use netplay. Instead a reference or pointer to the instances would passed into the core from the calling code, where the instances are then passed into relevant classes* and stored as a member variable. That's one solution anyway. I'm sure there's other better ways as well. It's also possible that the way netplay is currently handled should be altered to have a better interface, that would need more looking into than what's necessary to make a Github comment like this though

*This may or may not involve the creation of new classes or altering current class hierarchies. This should also be as least invasive as possible to classes.

Reply:
Yeah the actual instance creation should remain in the frontend, I just mean move the pointers. What if we move the pointers from `DolphinQt/Settings.h` to `Core/NetPlayProto.h`, inside the NetPlay namespace as a static variable? They would still be globally accessible, but no longer file-scoped and instead inside a qualified namespace, and I don't think de-globalizing access to the client/server instances makes sense since they're accessed in various places in the core.

Reply:
Well, Movie and NetPlay are both pretty messy, with bits and pieces of special cases for them littered all over the codebase. Rewriting that is well beyond the scope of this PR, and a pretty big job in itself, so I can't see any way to avoid doing things this way for now. I'd rather not have a pretty big feature blocked by something like a huge rewrite of NetPlay.

Reply:
> I'd rather not have a pretty big feature like this blocked by something like a huge refactor of NetPlay.

Honestly disappointing to hear making a pre-existing problem more of a pain to clean up before actually introducing a feature is a preferable choice, especially after you said you understood the concerns about the quoted interpretation I posted a bit ago, but at this point, however, it's not a shock to see said interpretation occur for the N+1th time; you do you.

Reply:
You do make a fair argument. Okay, how about this: I'll rewrite some of the changes here to have as little additional code smell as possible. I'll remove the additional globals, and add functionality to allow NetPlay to do what it needs in the relevant interfaces, in a manner that doesn't require special casing it and can work as a more general functionality.